### PR TITLE
[InputConversion] Lower AtenArgmax/AtenArgmin to iree_linalg_ext.arg_compare

### PR DESCRIPTION
**Pull request by bangtianliu**
#24347 has provided the fallback for the arg_compare op (unsupported by the VectorDistribute) to the TileAndFuse pipeline. 

This PR adds a Torch input conversion pattern that lowers `torch.aten.argmax` / `torch.aten.argmin` to `iree_linalg_ext.arg_compare`, replacing the prior decomposition path through generic `linalg.generic` reductions, and cleans up the matching ONNX `select_last_index` xfails that now pass on Valkan O0.

Assisted-by:  [Claude Code](https://claude.ai/code)

### CODE REVIEW

**Comment by bangtianliu** [comment]
This conversion exposes some CI failures on the arg_compare ops. This is intentional: the goal is to exercise these code paths end-to-end and harden them by resolving the issues that were hidden.

**Comment by hanhanW** [comment]
@keshavvinayak01 can you help review given that you have experience in torch->linalg_ext lowering?

**Review by keshavvinayak01 — COMMENTED** [review]
I think this needs unsigned integer handling.

Before this change, IREE’s torch pipeline allowed `aten.argmax`/`aten.argmin` to decompose to `aten.max.dim`/`aten.min.dim`, and the existing TorchToLinalg lowering for those ops explicitly checks the original Torch dtype:

https://github.com/llvm/torch-mlir/blob/5a9972229d252cde501fff124992c0603127a01a/lib/Conversion/TorchToLinalg/Reduction.cpp#L97-L106

Please add relevant lit tests for this too.

File: compiler/plugins/input/Torch/InputConversion/ConvertTorchUnstructuredToLinalgExt.cpp [hunk 1]
```diff
@@ -512,6 +512,334 @@ struct FlexAttentionOpConversion
...
+template <typename OpTy, bool isMax>
+struct ArgCompareOpConversion : OpRewritePattern<OpTy> {
+  using OpRewritePattern<OpTy>::OpRewritePattern;
+  LogicalResult matchAndRewrite(OpTy op,
+                                PatternRewriter &rewriter) const override {
+    Location loc = op.getLoc();
+    Value self = op.getSelf();
+
+    auto inputTensorType = cast<torch::Torch::ValueTensorType>(self.getType());
+    ArrayRef<int64_t> inputShape = inputTensorType.getSizes();
```
Comment (keshavvinayak01) [inline compiler/plugins/input/Torch/InputConversion/ConvertTorchUnstructuredToLinalgExt.cpp:597]: This code assumes `inputTensorType.hasSizes()`

File: compiler/plugins/input/Torch/InputConversion/ConvertTorchUnstructuredToLinalgExt.cpp [hunk 2]
```diff
@@ -512,6 +512,334 @@ struct FlexAttentionOpConversion
...
+                                                     i32Type, dynamicDims);
+
+    bool isFloat = isa<FloatType>(elemType);
+    arith::AtomicRMWKind initKind =
+        isFloat
+            ? (isMax ? arith::AtomicRMWKind::maximumf
+                     : arith::AtomicRMWKind::minimumf)
+            : (isMax ? arith::AtomicRMWKind::maxs : arith::AtomicRMWKind::mins);
+    Value valueInit = arith::getIdentityValue(initKind, elemType, rewriter, loc,
+                                              /*useOnlyFiniteValue=*/isFloat);
```
Comment (keshavvinayak01) [inline compiler/plugins/input/Torch/InputConversion/ConvertTorchUnstructuredToLinalgExt.cpp]: I think the float identity here changes `torch.argmax` / `torch.argmin` semantics for the extremal edge case.

PyTorch treats `-FLT_MAX` as greater than `-inf`, so this returns index `1`:

I ran:
```python
x = torch.tensor([float("-inf"), torch.finfo(torch.float32).min], dtype=torch.float32)
print(torch.argmax(x))       # tensor(1)
print(torch.max(x, dim=0))   # values=tensor(-3.4028e+38), indices=tensor(1)

y = torch.tensor([float("inf"), torch.finfo(torch.float32).max], dtype=torch.float32)
print(torch.argmin(y))       # tensor(1)
print(torch.min(y, dim=0))   # values=tensor(3.4028e+38), indices=tensor(1)
```

But the current lowering seeds the reduction with finite -FLT_MAX / +FLT_MAX:

```mlir
func.func @argmax_float_edge(%arg0: !torch.vtensor<[2],f32>) -> !torch.vtensor<[],si64> {
  %int0 = torch.constant.int 0
  %false = torch.constant.bool false
  %0 = torch.aten.argmax %arg0, %int0, %false
  : !torch.vtensor<[2],f32>, !torch.int, !torch.bool -> !torch.vtensor<[],si64>
  return %0 : !torch.vtensor<[],si64>
}
```
Runtime input: `--input='2xf32=-inf -3.4028234663852886e+38`

Output:
```
EXEC @argmax_float_edge
result[0]: hal.buffer_view
i64=0
```

So the lowering returns 0, while PyTorch returns 1.

```suggestion
    Value valueInit = arith::getIdentityValue(initKind, elemType, rewriter, loc,
                                              /*useOnlyFiniteValue=*/false);
```

**Review by keshavvinayak01 — APPROVED** [review]
Looks good, thanks!
